### ISSUE TITLE
About the metric.

### ISSUE THREAD

**Issue by zxczrx123** [Jabb0/FastFlow3D#8]
Hi,
Thanks for your implementation! Now I have a question about the calculation of the metric. In your code, I find you compute the pointwise metric at each step. Then, pytorch lightning will average the metric on each step automatically to get the mean metric on epoch. In my understanding, the point-wise mentioned in the paper is performed on the entire epoch. I want to know if this will lead to some bias in evaluation.

**Comment by Jabb0** [comment]
Hi,
that is a good finding.

We compute the loss over all entries of the batch (i.e. the all points in all point clouds). This is necessary for backpropagation in training.
PytorchLightning collects the output of each log for all steps in the epoch and reduces them at the end (by taking the mean). As the points clouds (and thus batches) are not the same size this is not the mean over all points in the epoch. It is the mean over all the batch means.
To achieve that one would need to log the number of points in the batch as well and perform a custom reduction on `epoch_end`.

That said I am not sure how the original authors did their metrics calculation based on the paper.
I think this is close if all batches have the roughly same amount of points. But not sure about it.

**Comment by zxczrx123** [comment]
@Jabb0 Thank you for your reply! Actually, I have achieved point-wise loss in the epoch by using Torchmetric. This does not make a big difference.
